### ISSUE TITLE
Running mlRedeploy or mlReloadModules are returning errors

### ISSUE THREAD

**Issue by peetkes** [marklogic/marklogic-data-hub#1462]
I've created a pristine newly created dhf v4.0.1 project and the initial deployment will run OK.
When I then try something like mlRedeploy or mlReloadModules I get an error like below:
````
./gradlew mlRedeploy -i
Initialized native services in: /Users/pkester/.gradle/native
Using 8 worker leases.
Starting Build
Settings evaluated using settings file '/Users/pkester/Documents/GitHub/dhf-test-v4/settings.gradle'.
Projects loaded. Root project using build file '/Users/pkester/Documents/GitHub/dhf-test-v4/build.gradle'.
Included projects: [root project 'dhf-test-v4']

> Configure project :
Evaluating root project 'dhf-test-v4' using build file '/Users/pkester/Documents/GitHub/dhf-test-v4/build.gradle'.
Compiling build file '/Users/pkester/Documents/GitHub/dhf-test-v4/build.gradle' using SubsetScriptTransformer.
PropertiesPlugin:apply Loaded 46 properties from /Users/pkester/Documents/GitHub/dhf-test-v4/gradle.properties
PropertiesPlugin:apply Loaded 4 properties from /Users/pkester/Documents/GitHub/dhf-test-v4/gradle-local.properties
PropertiesPlugin:apply Loaded 1 properties from /Users/pkester/.gradle/gradle.properties
PropertiesPlugin:apply Loaded 0 properties from environment variables
PropertiesPlugin:apply Loaded 0 properties from system properties
PropertiesPlugin:apply Loaded 0 properties from the command line

Initializing ml-gradle
Admin interface host: localhost
Admin interface username: admin
Admin interface scheme: https
Use simple SSL for Admin interface: true
App host: localhost
Using simple SSL context and 'ANY' hostname verifier for authenticating against the App-Services server
REST admin username: admin
App Services username: admin
Module permissions: rest-reader,read,rest-writer,insert,rest-writer,update,rest-extension-user,execute,data-hub-role,read,data-hub-role,execute
Manage host: localhost
Manage username: admin
Manage user with security role: admin
Manage scheme: https
Use simple SSL for Manage app server: true
Initializing ManageClient with manage config of: [ManageConfig host: localhost, port: 8002, username: admin, security username: admin]
Configuring simple SSL approach for connecting to: https://localhost:8002
Configuring simple SSL approach for connecting to: https://localhost:8001
Finished initializing ml-gradle


Initializing data-hub-gradle
Hub Project Dir: /Users/pkester/Documents/GitHub/dhf-test-v4
App host: localhost
Using simple SSL context and 'ANY' hostname verifier for authenticating against the App-Services server
REST admin username: admin
App Services username: admin
Module permissions: rest-reader,read,rest-writer,insert,rest-writer,update,rest-extension-user,execute,data-hub-role,read,data-hub-role,execute
Finished initializing ml-data-hub

Compiling build file '/Users/pkester/Documents/GitHub/dhf-test-v4/build.gradle' using BuildScriptTransformer.
All projects evaluated.
Selected primary task 'mlRedeploy' from project :
Tasks to be executed: [task ':mlClearModulesDatabase', task ':deleteModuleTimestampsFile', task ':mlDeployRoles', task ':mlDeployUsers', task ':mlDeploySecurity', task ':hubPreInstallCheck', task ':mlDeleteModuleTimestampsFile', task ':mlPrepareRestApiDependencies', task ':mlDeployApp', task ':mlPostDeploy', task ':mlDeploy', task ':mlRedeploy']
:mlClearModulesDatabase (Thread[Task worker for ':' Thread 3,5,main]) started.

> Task :mlClearModulesDatabase FAILED
Task ':mlClearModulesDatabase' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Clearing DHF modules from modules database
:mlClearModulesDatabase (Thread[Task worker for ':' Thread 3,5,main]) completed. Took 1.516 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':mlClearModulesDatabase'.
> Local message: User is not allowed to search. Server Message: Server (not a REST instance?) did not respond with an expected REST Error message.

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
1 actionable task: 1 executed
````
 
Other commands will run succesfully.
I'm running on a docker CentOS 7 container with ml9.0-7 with dhf 4.0.1 on MacOS High Sierra.

[stacktrace-clearModulesIssue.txt](https://github.com/marklogic/marklogic-data-hub/files/2519322/stacktrace-clearModulesIssue.txt)

**Comment by peetkes** [comment]
As an extra pointer: I configured all ports for SSL. When I disable SSL for all ports mlRedeploy and mlReloadModules will work!

**Comment by grechaw** [comment]
Thanks for that second bit...  how urgent is the SSL coverage/support?

**Comment by grechaw** [comment]
Assigning to @sbayatpur-ml to consider prioritization, as this is the first explicit need for SSL that has been reported.  There is a story internally to support SSL explicitly everywhere.  The tests currently cover running DHF on an SSL-enabled setup, but they also setup the systems first, *then* enable SSL, which is not what @peetkes is doing here.

**Comment by sbayatpur-ml** [comment]
@grechaw Thanks. Is https://project.marklogic.com/jira/browse/DHFPROD-239 the story you're referring to?

**Comment by damonfeldman** [comment]
@grechaw - does this mean that a workaround is to disable SSL, do the setup or changes, and re-enable SSL afterwards?

I think this may be ok in development, and even in production for cases where upgrades are done in an outage window.

**Comment by aebadirad** [comment]
This was resolved some time ago in 4.1.1. Closing out.
